cmd/main_postsubmit.go

```diff
@@ -1,128 +1,192 @@
 package main
 
 import (
+	"flag"
 	"fmt"
+	"io"
 	"log"
 	"os"
 	"os/exec"
-	"strconv"
+	"path/filepath"
 	"strings"
 )
 
+var (
+	outputStream io.Writer = os.Stdout
+	errStream    io.Writer = os.Stderr
+)
+
 type Command struct {
 	releaseBranch        string
 	gitRoot              string
 	release              string
 	artifactBucket       string
 	uploadToPublicBucket bool
-	args                 []string
+
+	makeTarget string
+	makeArgs   []string
+
+	dryRun bool
 }
 
-func (cmd Command) run(runMake bool, uploadArtifacts bool, targetPath string) {
-	if runMake == true {
-		// Add a specific target
-		cmd.args[1] = cmd.args[1] + targetPath
-		command := exec.Command("make", cmd.args...)
-		output, err := command.CombinedOutput()
+func (c *Command) runMake(uploadArtifacts bool, targetPath string) error {
```

Comment:
This function tries to make 2 non make exec calls. Probably, we can rename it to only run?

Reply:
yea, I can rename it